src/data_generators/publisher_data_file.py

```diff
@@ -0,0 +1,198 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base class for publisher data
+
+Represents real or simulated impression log data for a single publisher.
+"""
+
+from collections import Counter
+from copy import deepcopy
+from numpy.random import randint
+from os.path import join
+from typing import Dict
+from typing import Iterable
+from typing import Tuple
+from wfa_planning_evaluation_framework.data_generators.impression_generator import (
+    ImpressionGenerator,
+)
+from wfa_planning_evaluation_framework.data_generators.pricing_generator import (
+    PricingGenerator,
+)
+
+
+class PublisherDataFile:
```

Comment:
Are we sure we want to add "File" to this name? 
It seems to me PublisherData is better. You have "write_publisher_data_file" too, maybe we can change it to just "write" or "persist". The other methods : impressions, spend, etc... makes more sense with just PublisherData to me. Wdyt?

Reply:
I have changed PublisherDataFile to PublisherData.  I think I am going to keep "write" rather than change it to "persist", since "read" and "write" are normally used in pairs.  (What would you then change "read" to?  Depersist?  No thanks.)

Reply:
Sounds good.